src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java

```diff
@@ -0,0 +1,68 @@
+package edu.ucsb.cs156.team02.controllers;
+
+import edu.ucsb.cs156.team02.entities.CollegiateSubreddit;
+import edu.ucsb.cs156.team02.entities.User;
+import edu.ucsb.cs156.team02.models.CurrentUser;
+import edu.ucsb.cs156.team02.repositories.CollegiateSubredditRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Api(description = "CollegiateSubreddit")
+@RequestMapping("/api/collegiateSubreddits")
+@RestController
+@Slf4j
+public class CollegiateSubredditController extends ApiController {
+
+    @Autowired
+    CollegiateSubredditRepository collegiateSubredditRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "List all collegiate subreddits in database")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<CollegiateSubreddit> getCollegiateSubreddits() {
+        loggingService.logMethod();
+        Iterable<CollegiateSubreddit> collegiateSubs = collegiateSubredditRepository.findAll();
+        return collegiateSubs;
+    }
```

Comment:
I think this endpoint should not be accessible to `USERS`. I was reading [a thread](https://ucsb-cs156-w22.slack.com/archives/C030KT692A3/p1643670228956679?thread_ts=1643668881.909719&cid=C030KT692A3) by Prof. Conrad last night about this on Slack.

Reply:
Oh, I'm wrong. Dismissing.
> For our three database tables, though, all records are available to everyone.